### ISSUE TITLE
"semicolons" rule removing a semicolon which affects the behavior of the code

### ISSUE THREAD

**Issue by louis-prudhomme** [nicklockwood/SwiftFormat#1345]
Hi ! I use [TCA](https://github.com/pointfreeco/swift-composable-architecture)'s dependency management mechanism, which is detailed [here](https://github.com/pointfreeco/swift-composable-architecture/blob/4ece153d6b26a407fab036decacbdf7e869598f0/Sources/ComposableArchitecture/Documentation.docc/Articles/DependencyManagement.md).

Essentially, it consists in using the expresion `@Dependency(\.myDependency) var myDependency` where needed, after which `myDependency` will be injected and can be used. 

While it is primarily meant to be used as properties in `struct` and `class` declarations, I've recently come to use it in `func` bodies and in closures, which led to the following code:

```swift
struct ApiClient {
    static func live() -> Self {
        authenticate: { loginOption in
            await session.authenticate(with: loginOption)
                .mapError { error in
                    @Dependency(\.crashAnalyticsClient) var crashClient;
                    crashClient.track(error)
                    return error
                }
        }
    }
}
```

The `;` at the end of the dependency injection declaration is necessary, otherwise the Swift compiler panics and interprets the line that follows as the same expression (_ie_, as: `@Dependency(\.crashAnalyticsClient) var crashClient crashClient.track(error)`, which causes a compilation error.

However, when running `swiftformat . --swiftversion 5`, the semicolon is removed, leading to a compilation error. As per the [semicolon rule](https://github.com/nicklockwood/SwiftFormat/blob/master/Rules.md#semicolons), "semicolon is not removed if it would affect the behavior of the code". 

I'm not sure whether the semicolon should be necessary here, reporting just in case.

**Comment by mbrandonw** [comment]
FWIW this is a bug in the Swift compiler and it was even fixed recently so it may not be an issue in a near future Swift: https://github.com/apple/swift-format/issues/439#issuecomment-1301502584

**Comment by louis-prudhomme** [comment]
Thanks ! I'll use the workaround and specify explicit types for the time being until the fix gets shipped!

**Comment by nicklockwood** [comment]
@louis-prudhomme I've added a fix for this in SwiftFormat 0.50.8, so your workaround shouldn't be needed
